upgrade from pysam-0.9.1.2 to pysam-0.11.2.2 fails

Attached the full terminal log during upgrade  (triggered by installing CrossMap)

Thanks for your help, this error is now on two independent macs with different OS'es (Yosemite and ElCapitan). lzma.h seem to be missing!! but lzma is installed and the .h is present in my macport include folder

```
ll /opt/local/include/lzma.h 
-rw-r--r-- 1 root admin 9.6K Dec 31  2016 /opt/local/include/lzma.h
```

```
lzma --version
xz (XZ Utils) 5.2.3
liblzma 5.2.3
```

Best

[pysam_upgrade_failed.txt](https://github.com/pysam-developers/pysam/files/1117873/pysam_upgrade_failed.txt)

I am running into a similar issue now with Travis CI.
```
E   ImportError: dlopen(/Users/travis/miniconda3/envs/testenv/lib/python2.7/site-packages/pysam-0.12.0.1-py2.7-macosx-10.5-x86_64.egg/pysam/libchtslib.so, 2): Library not loaded: @rpath/liblzma.5.dylib

E     Referenced from: /Users/travis/miniconda3/envs/testenv/lib/python2.7/site-packages/pysam-0.12.0.1-py2.7-macosx-10.5-x86_64.egg/pysam/libchtslib.so

E     Reason: Incompatible library version: libchtslib.so requires version 8.0.0 or later, but liblzma.5.dylib provides version 6.0.0[0m
```
This used to work fine and stopped working recently after an unrelated code change.

--- a/devtools/run_tests_travis.sh
+++ b/devtools/run_tests_travis.sh
@@ -43,7 +43,7 @@ conda install -y "samtools=1.9" "bcftools=1.9" "htslib=1.9" xz curl bzip2 conda-
 # Need to make C compiler and linker use the anaconda includes and libraries:
 export PREFIX=~/miniconda3/
 export CFLAGS="-I${PREFIX}/include -L${PREFIX}/lib"
-export HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl --disable-lzma"
+export HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"
 
 echo "show samtools, htslib, and bcftools versions"
 samtools --version
@@ -54,8 +54,10 @@ bcftools --version
 ~/miniconda3/bin/conda-build devtools/conda-recipe/ --python=$CONDA_PY
 
 # install code from the repository via setup.py
-echo "installing via setup.py from repository"
-python setup.py install
+echo
+echo "============ installing via setup.py from repository ============"
+echo
+python setup.py install || exit
 
 # create auxilliary data
 echo

--- a/htslib/configure
+++ b/htslib/configure
@@ -4751,12 +4751,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_deflate_libdeflate_deflate_compress" >&5
 $as_echo "$ac_cv_lib_deflate_libdeflate_deflate_compress" >&6; }
 if test "x$ac_cv_lib_deflate_libdeflate_deflate_compress" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBDEFLATE 1
-_ACEOF
-
-  LIBS="-ldeflate $LIBS"
-
+  :
 else
   libdeflate='missing library'
 fi
@@ -4765,6 +4760,7 @@ fi
 
 $as_echo "#define HAVE_LIBDEFLATE 1" >>confdefs.h
 
+     LIBS="-ldeflate $LIBS"
      private_LIBS="$private_LIBS -ldeflate"
      static_LIBS="$static_LIBS -ldeflate"
 else

--- a/htslib/configure.ac
+++ b/htslib/configure.ac
@@ -275,9 +275,10 @@ fi
 AS_IF([test "x$with_libdeflate" != "xno"],
   [libdeflate=ok
    AC_CHECK_HEADER([libdeflate.h],[],[libdeflate='missing header'],[;])
-   AC_CHECK_LIB([deflate], [libdeflate_deflate_compress],[],[libdeflate='missing library'])
+   AC_CHECK_LIB([deflate], [libdeflate_deflate_compress],[:],[libdeflate='missing library'])
    AS_IF([test "$libdeflate" = "ok"],
     [AC_DEFINE([HAVE_LIBDEFLATE], 1, [Define if libdeflate is available.])
+     LIBS="-ldeflate $LIBS"
      private_LIBS="$private_LIBS -ldeflate"
      static_LIBS="$static_LIBS -ldeflate"],
     [AS_IF([test "x$with_libdeflate" != "xcheck"],

--- a/setup.py
+++ b/setup.py
@@ -51,6 +51,7 @@ def changedir(path):
 
 
 def run_configure(option):
+    sys.stdout.flush()
     try:
         retcode = subprocess.call(
             " ".join(("./configure", option)),
@@ -78,6 +79,25 @@ def run_make_print_config():
     return make_print_config
 
 
+@contextmanager
+def set_compiler_envvars():
+    tmp_vars = []
+    for var in ['CC', 'CFLAGS', 'LDFLAGS']:
+        if var in os.environ:
+            print ("# pysam: (env) {}={}".format(var, os.environ[var]))
+        elif var in sysconfig.get_config_vars():
+            value = sysconfig.get_config_var(var)
+            print ("# pysam: (sysconfig) {}={}".format(var, value))
+            os.environ[var] = value
+            tmp_vars += [var]
+
+    try:
+        yield
+    finally:
+        for var in tmp_vars:
+            del os.environ[var]
+
+
 def configure_library(library_dir, env_options=None, options=[]):
 
     configure_script = os.path.join(library_dir, "configure")
@@ -91,7 +111,7 @@ def configure_library(library_dir, env_options=None, options=[]):
         raise ValueError(
             "configure script {} does not exist".format(configure_script))
 
-    with changedir(library_dir):
+    with changedir(library_dir), set_compiler_envvars():
         if env_options is not None:
             if run_configure(env_options):
                 return env_options

